daemon/cmd/daemon.go

```diff
@@ -188,15 +188,14 @@ func (d *Daemon) init() error {
 		}
 
 		if option.Config.SockopsEnable {
-			disableSockops := func(err error) {
-				option.Config.SockopsEnable = false
-				log.WithError(err).Warn("Disabled '--sockops-enable' due to missing BPF kernel support")
+			errSockops := func(err error) {
```

Comment:
I think we can simplify this to avoid the function call given that it's now a simple error.